variables.tf

```diff
@@ -54,8 +54,14 @@ variable "admin_password" {
   type        = string
 }
 
+variable "admin_ssh_key_algorithm" {
+  description = "The algorithm to use for the admin SSH key pair. Valid values are `RSA` or `ED25519`."
+  default     = "ED25519"
+  type        = string
+}
+
```

Comment:
Update name and description to indicate that this variable is only used when `admin_ssh_public_key` is not set, triggering SSH key generation when `authentication_type` is `SSH`.